engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -112,10 +114,25 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
       writeToMetadata(Map(callCachingReadResultMetadataKey -> "Cache Miss"))
       log.info(s"Cache miss for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}")
       runJob(jobDescriptor, bjeaProps)
-    case Event(CacheHit(cacheResultId), JobDescriptorData(jobDescriptor, _)) =>
+    case Event(CacheHit(cacheResultId), JobDescriptorData(jobDescriptor, bjeaProps)) =>
       writeToMetadata(Map(callCachingReadResultMetadataKey -> s"Cache Hit (from result ID $cacheResultId)"))
       log.info(s"Cache hit for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}! Copying cache result $cacheResultId")
-      lookupCachedResult(jobDescriptor, cacheResultId)
+      lookupCacheResult(jobDescriptor, bjeaProps, jobKey.call.task.outputs, cacheResultId)
+  }
+
+  // When PreparingCachedOutputs, the FSM should have EJEAJobDescriptorData
+  when(PreparingCachedOutputs) {
+    case Event(CachedOutputLookupSucceeded(cachedJobOutputs), JobDescriptorData(jobDescriptor, _)) =>
+      //I can remove some of this logging once this PR is reviewed--mostly for my own debugging
+      log.info(s"Created a copy of the cached job outputs for $jobTag")
+      copyCacheHit(jobDescriptor, cachedJobOutputs)
+    case Event(CachedOutputLookupFailed(metaInfoId, error), JobDescriptorData(jobDescriptor, bjeaProps)) =>
+      log.info(s"Can't make a copy of the cached job outputs for $jobTag due to $error. Running job.")
+      runJob(jobDescriptor, bjeaProps)
+    case Event(_: CallCacheHashes, _) =>
+      // TODO MLC is this really happening?  How would this be possible?
+      log.info(s"Getting these hashes when cache hit already found. Waiting for CachedJobOutputs")
```

Comment:
This situation *might* be possible - we might have hashes which are not required for determining cache hit (e.g. the memory runtime attribute). We store them regardless.

Regardless, the `CallCacheHashes` are sent to the `EJEA` **after** the `Hit`/`Miss` so it's quite possible there'd be a race condition which slows down the sending of the second message until now.